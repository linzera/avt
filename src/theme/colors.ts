const blacks = {
  black12: 'rgba(0, 0, 0, 0.12)',
  black38: 'rgba(0, 0, 0, 0.38)',
  black60: 'rgba(0, 0, 0, 0.60)',
  black80: 'rgba(0, 0, 0, 0.80)',
  black100: 'rgb(0, 0, 0)',
};

const whites = {
  white12: 'rgba(255, 255, 255, 0.12)',
  white30: 'rgba(255, 255, 255, 0.30)',
  white70: 'rgba(255, 255, 255, 0.70)',
  white100: '#ffffff',
};

const grays = {
  gray50: '#fafafa',
  gray100: '#f5f5f5',
  gray200: '#eeeeee',
  gray300: '#e0e0e0',
  gray400: '#bdbdbd',
  gray500: '#9e9e9e',
  gray600: '#757575',
  gray800: '#424242',
};

const blueGrays = {
  blueGray50: '#eceff1',
  blueGray100: '#cfd8dc',
  blueGray500: '#607d8b',
  blueGray600: '#546e7a',
  blueGray800: '#37474f',
};

const reds = {
  red50: '#feede7',
  red100: '#fdc9b9',
  red500: '#f83c00',
  red600: '#e23700',
  red800: '#b52c00',
};

const greens = {
  green50: '#f3f8ed',
  green100: '#dbeacb',
  green500: '#7cb342',
  green600: '#669337',
  green800: '#4f722a',
};

const yellows = {
  yellow50: '#fefae8',
  yellow100: '#fdf0bb',
  yellow500: '#f9ca06',
  yellow600: '#e3b806',
  yellow800: '#cca605',
};

const blues = {
  blue50: '#e7f5fd',
  blue100: '#b9e1f9',
  blue500: '#0091ea',
  blue600: '#0077c0',
  blue800: '#005d95',
};

const oranges = {
  orange50: '#fef4e7',
  orange100: '#fddfb9',
  orange500: '#fb8c00',
  orange600: '#e58000',
  orange800: '#ce7300',
};

const colors = {
  ...blacks,
  ...whites,
  ...grays,
  ...blueGrays,
  ...reds,
  ...greens,
  ...yellows,
  ...blues,
  ...oranges,
  androidRippleColor: 'rgba(0, 0, 0, .32)',
};

export type ColorType = typeof colors;
export type Color = keyof ColorType;

export default colors;
