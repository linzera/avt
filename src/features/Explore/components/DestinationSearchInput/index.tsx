import React, {useState} from 'react';
import {
  NativeSyntheticEvent,
  TextInput,
  TextInputFocusEventData,
  TextInputProps,
  View,
} from 'react-native';
import Button from '~/components/Button';
import colors from '~/theme/colors';

import SearchIcon from '@explore/assets/svg/search_icon.svg';
import ClearIcon from '@explore/assets/svg/clear_icon.svg';

interface Props extends TextInputProps {
  onClear: () => void;
}

export default function DestinationSearchInput({
  value,
  onFocus,
  onBlur,
  onClear,
  ...rest
}: Props) {
  const [isFocused, setFocus] = useState(false);

  function onFocusText(e: NativeSyntheticEvent<TextInputFocusEventData>) {
    if (onFocus) {
      onFocus(e);
    }

    setFocus(true);
  }

  function onBlurText(e: NativeSyntheticEvent<TextInputFocusEventData>) {
    if (onBlur) {
      onBlur(e);
    }

    setFocus(false);
  }

  return (
    <View
      style={{
        borderBottomWidth: 1,
        marginBottom: 20,
        marginHorizontal: 20,
        marginTop: 8,
        paddingBottom: 4,
        borderColor: colors.gray300,
        height: 50,
        flexDirection: 'row',
        alignItems: 'center',
        ...(isFocused ? {borderColor: colors.accent} : {}),
      }}>
      <SearchIcon />
      <TextInput
        style={{
          flex: 1,
          height: '100%',
          color: 'black',
          marginLeft: 8,
          fontSize: 16,
        }}
        value={value}
        onFocus={onFocusText}
        onBlur={onBlurText}
        placeholder="Search by destination name"
        placeholderTextColor="#022B5440"
        {...rest}
      />
      {value ? (
        <Button onPress={onClear} variant="icon">
          <ClearIcon />
        </Button>
      ) : undefined}
    </View>
  );
}
