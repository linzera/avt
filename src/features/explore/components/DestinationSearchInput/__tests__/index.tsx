import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import colors from '~/theme/colors';
import DestinationSearchInput from '..';

describe('<DestinationSearchInput />', () => {
  it('should render as expected given default props', () => {
    const {toJSON, queryByHintText} = render(
      <DestinationSearchInput onClear={() => undefined} />,
    );

    expect(queryByHintText('Clear Button')).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should behave as expected on focus and blur', async () => {
    const onFocus = jest.fn();
    const onBlur = jest.fn();

    const {getByHintText, getByTestId} = render(
      <DestinationSearchInput
        onClear={() => undefined}
        onBlur={onBlur}
        onFocus={onFocus}
      />,
    );

    const inputHintText = 'Search Destination Input';

    await fireEvent(getByHintText(inputHintText), 'focus');

    expect(onFocus).toHaveBeenCalled();
    expect(getByTestId('SearchInputContainer')).toHaveStyle({
      borderColor: colors.accent,
    });

    await fireEvent(getByHintText(inputHintText), 'blur');

    expect(onBlur).toHaveBeenCalled();
    expect(getByTestId('SearchInputContainer')).toHaveStyle({
      borderColor: colors.gray300,
    });
  });

  it('should render clear button given valid value', async () => {
    const onClear = jest.fn();

    const {queryByHintText, getByHintText} = render(
      <DestinationSearchInput onClear={onClear} value="Value" />,
    );

    expect(queryByHintText('Clear Input')).not.toBeNull();

    await fireEvent.press(getByHintText('Clear Input'));

    expect(onClear).toHaveBeenCalledWith();
  });
});
