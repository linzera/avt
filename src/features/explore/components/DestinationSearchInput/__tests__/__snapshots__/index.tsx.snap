// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DestinationSearchInput /> should render as expected given default props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#e0e0e0",
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 8,
        "paddingBottom": 4,
      },
      false,
    ]
  }
  testID="SearchInputContainer"
>
  <SvgMock />
  <TextInput
    accessibilityHint="Search Destination Input"
    allowFontScaling={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Search by destination name"
    placeholderTextColor="#022B5440"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#505051",
        "flex": 1,
        "fontSize": 16,
        "height": "100%",
        "marginLeft": 8,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
